### PR TITLE
add workflow to test external-dns PRs

### PR DESCRIPTION
**Pull request by adobley**
## What this PR does / why we need it

On changes to external-dns files the unit tests will be run. This gives more confidence on PRs for external-dns.

The tests were failing to run due to a go.mod issue, so there is an update to the dependencies.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
NONE
```

## Describe testing done for PR

Tested in a fork to verify the workflow behaves correctly.

### CODE REVIEW

**Review by mcwumbly — DISMISSED** [review]
lgtm

I see the tests ran successfully in this PR here: https://github.com/vmware-tanzu/community-edition/runs/4958766271?check_suite_focus=true#step:5:75

**Review by jpmcb — COMMENTED** [review]
This looks good to me! Will also let @dvonthenen get eyes on this since he's done alot of the work in github actions.

File: .github/workflows/test-pr-external-dns.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+on:
+  pull_request:
+    types:
+      - ready_for_review
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - "addons/packages/external-dns/**/*"
```
Comment (dvonthenen) [inline .github/workflows/test-pr-external-dns.yaml:11]: I would also add modification of this file/action itself in the path list

**Review by davidvonthenen — COMMENTED** [review]
Just left the one comment. Looks pretty straightforward!

File: .github/workflows/test-pr-external-dns.yaml (hunk 1 above)
Comment (christianang) [reply .github/workflows/test-pr-external-dns.yaml:11]: That makes sense. I've added it to the paths.

**Review by jpmcb — APPROVED** [review]

**Pull request merged**